core/trino-main/src/test/java/io/trino/type/TestMapOperatorsLegacy.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.type;
+
+import io.airlift.slice.Slice;
+import io.trino.operator.scalar.AbstractTestFunctions;
+import io.trino.spi.function.LiteralParameters;
+import io.trino.spi.function.ScalarFunction;
+import io.trino.spi.function.SqlType;
+import io.trino.spi.type.StandardTypes;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static io.trino.testing.DateTimeTestingUtils.sqlTimestampOf;
+import static io.trino.type.JsonType.JSON;
+import static java.lang.String.format;
+
+public class TestMapOperatorsLegacy
+        extends AbstractTestFunctions
+{
+    @BeforeClass
+    public void setUp()
+    {
+        registerScalar(getClass());
+    }
+
+    @ScalarFunction
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.JSON)
+    public static Slice uncheckedToJson(@SqlType("varchar(x)") Slice slice)
+    {
+        return slice;
+    }
+
+    @Test
+    public void testMapToJson()
+    {
+        // Test key ordering
+        assertFunction("CAST(MAP(ARRAY[7,5,3,1], ARRAY[8,6,4,2]) AS JSON)", JSON, "{\"1\":2,\"3\":4,\"5\":6,\"7\":8}");
+        assertFunction("CAST(MAP(ARRAY[1,3,5,7], ARRAY[2,4,6,8]) AS JSON)", JSON, "{\"1\":2,\"3\":4,\"5\":6,\"7\":8}");
+
+        // Test null value
+        assertFunction("cast(cast (null as MAP<BIGINT, BIGINT>) AS JSON)", JSON, null);
```

Comment:
Don't use angle brackets for parametric types. That's the legacy form. Use `map(bigint, bigint)`, instead.